lib/reservation.rb

```diff
@@ -0,0 +1,43 @@
+require "date"
+require "pry"
+
+class Reservation
+  attr_accessor :reservation_id, :check_in, :check_out, :room, :is_hotel_blocker
+
+  def initialize(reservation_id, check_in: nil, check_out: nil, room: nil, is_hotel_blocker: false)
+    @reservation_id = reservation_id
+    check_in == nil ? @check_in = nil : @check_in = Date.parse(check_in)
+    check_out == nil ? @check_out = nil : @check_out = Date.parse(check_out)
+    @room = all_rooms.sample
+    @is_hotel_blocker = is_hotel_blocker
+  end
+
+  def duration
+    if self.check_in != nil && self.check_out != nil
+      duration = (check_out - check_in).to_i
+
+      if duration <= 0
```

Comment:
Instead of calculating a duration as an `Integer` you can just compare the dates directly.

```suggestion
      if check_out <= check_in
```